lib/core/services/user-preferences.service.ts

```diff
@@ -70,7 +70,7 @@ export class UserPreferencesService {
             const locale = this.locale || this.getDefaultLocale();
 
             this.setWithoutStore(UserPreferenceValues.Locale, locale);
-            this.setWithoutStore('textOrientation', (this.locale || this.getDefaultLocale()));
+            this.setWithoutStore('textOrientation', this.getLanguageByKey(locale).direction || 'ltr');
```

Comment:
Accessing "direction" will cause error if "getLanguageByKey" does not find anything?

Reply:
we have [<LanguageItem> { key: 'en' }] as fallbacks and 'ltr' for undefined direction